File: app/src/main/java/com/example/dht11test2/ListData.java
package com.example.dht11test2;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.ListAdapter;
import android.widget.ListView;
import android.widget.SimpleAdapter;

import java.util.ArrayList;
import java.util.HashMap;

public class ListData extends AppCompatActivity {
    private ListView lv;

    ArrayList<HashMap<String, String>> final_data;

    ListData(){}

    ListData(ArrayList<HashMap<String, String>> final_data){
        this.final_data = final_data;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_list_data);

        lv = (ListView) findViewById(R.id.list);
        /**
         * Updating parsed JSON data into ListView
         * */
        ListAdapter adapter = new SimpleAdapter(
                ListData.this, Utils.final_data,
                R.layout.list_item, new String[]{"device_id", "user_id","temperature",
                "humidity"}, new int[]{R.id.name,
                R.id.device, R.id.temp, R.id.humi});

        lv.setAdapter(adapter);
    }
}

File: app/src/main/java/com/example/dht11test2/ListData.kt
package com.example.dht11test2

import android.os.Bundle
import android.view.View
import android.widget.ListAdapter
import android.widget.ListView
import android.widget.SimpleAdapter
import androidx.appcompat.app.AppCompatActivity
import java.util.*

class ListData() : AppCompatActivity() {


    private var lv: ListView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_list_data)
        lv = findViewById<View>(R.id.list) as ListView
        /**
         * Updating parsed JSON data into ListView
         */
        val adapter: ListAdapter = SimpleAdapter(
                this@ListData, Utils.final_data,
                R.layout.list_item, arrayOf("device_id", "user_id", "temperature",
                "humidity"), intArrayOf(R.id.name,
                R.id.device, R.id.temp, R.id.humi))
        lv!!.adapter = adapter
    }
}